jiant/proj/main/modeling/primary.py

```diff
@@ -291,6 +291,60 @@ def get_feat_spec(self, max_seq_length):
         )
 
 
+@JiantTransformersModelFactory.register(ModelArchitectures.DEBERTA)
+class JiantDebertaModel(JiantTransformersModel):
+    def __init__(self, baseObject):
+        super().__init__(baseObject)
+
+    @classmethod
+    def normalize_tokenizations(cls, tokenizer, space_tokenization, target_tokenization):
+        """See tokenization_normalization.py for details"""
+        breakpoint()
+        space_tokenization = [token.lower() for token in space_tokenization]
+        modifed_space_tokenization = bow_tag_tokens(space_tokenization)
+        modifed_target_tokenization = _process_sentencepiece_tokens(target_tokenization)
+
+        return modifed_space_tokenization, modifed_target_tokenization
+
+    def encode(self, input_ids, segment_ids, input_mask, output_hidden_states=True):
+        output = self.forward(
+            input_ids=input_ids,
+            token_type_ids=segment_ids,
+            attention_mask=input_mask,
+            output_hidden_states=output_hidden_states,
+        )
+        return JiantModelOutput(
+            pooled=output.last_hidden_state[:, 0, :],
+            unpooled=output.last_hidden_state,
+            other=output.hidden_states,
+        )
+
+    def get_mlm_weights_dict(self, weights_dict):
+        mlm_weights_map = {
+            "lm_predictions.lm_head.bias": "cls.predictions.bias",
+            "lm_predictions.lm_head.dense.weight": "cls.predictions.transform.dense.weight",
+            "lm_predictions.lm_head.dense.bias": "cls.predictions.transform.dense.bias",
+            "lm_predictions.lm_head.LayerNorm.weight": "cls.predictions.transform.LayerNorm.weight",
+            "lm_predictions.lm_head.LayerNorm.bias": "cls.predictions.transform.LayerNorm.bias",
+        }
+        mlm_weights_dict = {new_k: weights_dict[old_k] for new_k, old_k in mlm_weights_map.items()}
```

Comment:
Does `lm_predictions` come from somewhere? The dict here should look like BERT's.

Reply:
Looking at this again, I'm not sure where I came up with these keys. I've committed a corrected version.